### PR TITLE
[FAL-774]: new Analytics module and ES/SQL improvements

### PR DESCRIPTION
**Pull request by eLRuLL**
This PR includes:

- "analytics" module, which setups every AWS resource [in this guide](https://github.com/open-craft/openedx-deployment/blob/master/docs/analytics/AWS_setup.md)
- Makes the ElasticSearch module more customizable
- Adds ability for the SQL module to include security groups to the RDS instance

**Reviewers:**
- [ ] @pomegranited

### CODE REVIEW

File: modules/services/sql/variables.tf [hunk 1]
```diff
@@ -10,6 +10,10 @@ variable "max_allocated_storage" {
   default = 100
 }
 
+variable "extra_security_group_ids" {
```
Comment (pomegranited) [inline modules/services/sql/variables.tf:13]: Please add these new input variables to [sql/README](https://github.com/open-craft/terraform-scripts/blob/635890d9fb48661c6f531df494b1042bd23e5f1d/modules/services/sql/README.md#input)

File: modules/services/elasticsearch/variables.tf [hunk 2]
```diff
@@ -10,3 +10,26 @@ variable "number_of_nodes" {
 variable "create_iam_service_linked_role" {
   default = true
 }
+
+variable "zone_awareness_enabled" {
```
Comment (pomegranited) [inline modules/services/elasticsearch/variables.tf:14]: Please add these new input variables to [elasticsearch/README](https://github.com/open-craft/terraform-scripts/tree/635890d9fb48661c6f531df494b1042bd23e5f1d/modules/services/elasticsearch).

(Might be able to remove `number_of_subnets` and `start_number_of_subnets` depending on what you think about [this comment](https://github.com/open-craft/terraform-scripts/pull/18/files#diff-0f9cb3263848eba15ae3facdaa4b00ff2365de9c827c879aa4678d49855f6688R31)).

File: modules/services/elasticsearch/main.tf [hunk 3]
```diff
@@ -14,24 +14,21 @@ resource aws_elasticsearch_domain "openedx" {
...
+      availability_zone_count = var.availability_zone_count
     }
   }
 
   vpc_options {
-    subnet_ids = [
-      tolist(data.aws_subnet_ids.default.ids)[1],
-      tolist(data.aws_subnet_ids.default.ids)[2],
-    ]
+    subnet_ids = slice(tolist(data.aws_subnet_ids.default.ids), var.start_number_of_subnets, var.number_of_subnets)
```
Comment (pomegranited) [inline modules/services/elasticsearch/main.tf]: Are we ever going to use a subset of the `data.aws_subnet_ids` here? If so, then we're unlikely to be able to guarantee that the subnets we want are sequential in this list.

So I'd say either:
* Use all the subnets in `data.aws_subnet_ids`, OR
* Make the `subnet_ids` an input variable.

**Review by pomegranited — CHANGES_REQUESTED** [review]
@eLRuLL  This is looking great!
I haven't fully verified the analytics setup yet (since our client's analytics isn't yet running on their region), but the changes look good enough to merge, and we can iterate on the analytics stuff when we figure out what's not working there.

I've noted a few minor changes -- mostly docs, and one thing to simplify. But I'm good to approve once they're addressed.

Thank you for doing this! It's been needed for a long time.

File: modules/services/elasticsearch/main.tf (hunk 3 above)
Comment (eLRuLL) [reply modules/services/elasticsearch/main.tf]: @pomegranited you are right :+1: I set all subnets by default and now you can specify the subnet IDs as well

File: modules/services/elasticsearch/variables.tf (hunk 2 above)
Comment (eLRuLL) [reply modules/services/elasticsearch/variables.tf:14]: added :+1:

**Comment by eLRuLL** [comment]
@pomegranited  this should be ready for you review again, thanks

File: modules/services/sql/variables.tf (hunk 1 above)
Comment (eLRuLL) [reply modules/services/sql/variables.tf:13]: done :+1:

**Comment by pomegranited** [comment]
:+1: Ready to squash and merge @eLRuLL , thank you for adding this!

- [x] I tested this:
   * Insights and the analytics tasks work using the resources created by this change.
   * Ran `terraform plan` for this client, and saw no changes needed. 
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike>
- [x] Includes documentation
- [x] I made sure any change in configuration variables is reflected in the corresponding
    client's `configuration-secure` repository.

**Pull request merged**